core/include/drivers/tpm2/tpm2_cmd.h

```diff
@@ -0,0 +1,44 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2022, Linaro Limited
+ */
+
+#ifndef __DRIVERS_TPM2_CMD_H
+#define __DRIVERS_TPM2_CMD_H
+
+#include <io.h>
+#include <stdint.h>
+
+struct tpm2_cmd_hdr {
+	uint16_t tag;
+	uint32_t size;
+	union {
```

Comment:
Isn't this overkill? `uint32_t code;` could be used instead, it's only supposed to be accessed by special functions anyway.

Reply:
Ok..TPM has lots of structures...I kept a union just to avoid confusion..But since we are accessing it via special functions I will remove the union